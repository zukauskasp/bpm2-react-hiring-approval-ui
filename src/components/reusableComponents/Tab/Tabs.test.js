import React from 'react';
import TabsComponent from "./Tabs";
import { createShallow } from '@material-ui/core/test-utils';


describe('TabsComponent', () => {

    let shallow;
    let wrapper;

    beforeAll(() => {
        shallow = createShallow();
    });

    const firstTab = () => {
        return (
            <div>First tab</div>
        )
    }

    const SecondTab = () => {
        return (
            <div>Second tab</div>
        )
    }

    const tabConfig = [
        {
            tabName: "First tab",
            component: firstTab
        },
        {
            tabName: "Second tab",
            component: SecondTab
        }
    ];


    beforeEach(() => {
        wrapper = shallow(<TabsComponent tabConfig={tabConfig} taskId="1" />);
    })

    it('expect to match latest snapshot', () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('expect to render Tabs component', () => {
        expect(wrapper.length).toEqual(1);
    });


    it('expect to render correct tabs', () => {
        expect(wrapper.find({ label: tabConfig[0].tabName }).length).toEqual(1);
        expect(wrapper.find({ label: tabConfig[1].tabName }).length).toEqual(1);
        expect(wrapper.find({ label: "non existing label" }).length).toEqual(0);
    });

    it('expect to render correct component for given tab index', () => {
        expect(wrapper.find('TabPanel').find({ index: 0 }).find("firstTab").debug()).toEqual('<firstTab taskId="1" />');
        expect(wrapper.find('TabPanel').find({ index: 1 }).find("firstTab").length).toEqual(0);
    });

});