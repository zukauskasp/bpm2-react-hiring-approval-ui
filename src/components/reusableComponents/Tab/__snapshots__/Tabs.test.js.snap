// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsComponent expect to match latest snapshot 1`] = `
"<div className=\\"makeStyles-root-1\\">
  <WithStyles(ForwardRef(AppBar)) position=\\"static\\" className=\\"makeStyles-appBar-2\\">
    <WithStyles(ForwardRef(Tabs)) classes={{...}} value={0} onChange={[Function: handleChange]} aria-label=\\"simple tabs example\\">
      <WithStyles(ForwardRef(Tab)) classes={{...}} label=\\"First tab\\" />
      <WithStyles(ForwardRef(Tab)) classes={{...}} label=\\"Second tab\\" />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(AppBar))>
  <TabPanel value={0} index={0}>
    <firstTab taskId=\\"1\\" />
  </TabPanel>
  <TabPanel value={0} index={1}>
    <SecondTab taskId=\\"1\\" />
  </TabPanel>
</div>"
`;
